src/Types/BigNum.js

```diff
@@ -13,16 +13,57 @@ exports.zero = lib.BigNum.zero();
 
 exports.one = lib.BigNum.one();
 
+const add = (a, b) => {
+  return a + b;
+};
+
+const mul = (a, b) => {
+  return a * b;
+};
+
+const check_limit_with_func = (a, b, limit, func) => {
+  const r = func(a, b);
+  if (r < limit) {
+    return r;
+  } else {
+    throw new Error("Overflow detected");
+  }
+};
+
+const check_limit = (str, limit) => {
+  if (0 <= BigInt(str) && BigInt(str) < limit) {
+    return BigInt(str);
+  } else {
+    throw new Error("Overflow detected");
+  }
+};
+
 exports.bnAdd = maybe => lhs => rhs => {
+  // this is needed because try/catch overuse breaks runtime badly
+  // https://github.com/Plutonomicon/cardano-transaction-lib/issues/875
   try {
+    check_limit_with_func(
+      lhs.to_str(),
+      rhs.to_str(),
+      BigInt("18446744073709551616"),
```

Comment:
What is the significance of this number? I can guess, but it should be bound to a variable with a descriptive name. The literal is repeated as well in different places, which makes all of the checks more susceptible to typos